src/epd2in13_v2/mod.rs

```diff
@@ -0,0 +1,558 @@
+                    gs_trans: BorderWaveFormGS::LUT3,
+                },
+            )?;
+
+            self.set_vcom_register(spi, (-21).vcom())?;
+
+            self.set_gate_driving_voltage(spi, 190.gate_driving_decivolt())?;
+            self.set_source_driving_voltage(
+                spi,
+                150.source_driving_decivolt(),
+                50.source_driving_decivolt(),
+                (-150).source_driving_decivolt(),
+            )?;
+
+            self.cmd_with_data(spi, Command::SET_GATE_LINE_WIDTH, &[0x0A])?;
+
+            self.set_lut(spi, Some(self.refresh))?;
+        }
+
+        self.wait_until_idle();
+        Ok(())
+    }
+}
+
+impl<SPI, CS, BUSY, DC, RST> WaveshareDisplay<SPI, CS, BUSY, DC, RST>
+    for EPD2in13<SPI, CS, BUSY, DC, RST>
+where
+    SPI: Write<u8>,
+    CS: OutputPin,
+    BUSY: InputPin,
+    DC: OutputPin,
+    RST: OutputPin,
+{
+    fn new<DELAY: DelayMs<u8>>(
+        spi: &mut SPI,
+        cs: CS,
+        busy: BUSY,
+        dc: DC,
+        rst: RST,
+        delay: &mut DELAY,
+    ) -> Result<Self, SPI::Error> {
+        let mut epd = EPD2in13 {
+            interface: DisplayInterface::new(cs, busy, dc, rst),
+            sleep_mode: DeepSleepMode::MODE_1,
+            background_color: DEFAULT_BACKGROUND_COLOR,
+            refresh: RefreshLUT::FULL,
+        };
+
+        epd.init(spi, delay)?;
+        Ok(epd)
+    }
+
+    fn wake_up<DELAY: DelayMs<u8>>(
+        &mut self,
+        spi: &mut SPI,
+        delay: &mut DELAY,
+    ) -> Result<(), SPI::Error> {
+        self.init(spi, delay)
+    }
+
+    fn sleep(&mut self, spi: &mut SPI) -> Result<(), SPI::Error> {
+        self.wait_until_idle();
+
+        // All sample code enables and disables analog/clocks...
+        self.set_display_update_control_2(
+            spi,
+            DisplayUpdateControl2::new()
+                .enable_analog()
+                .enable_clock()
+                .disable_analog()
+                .disable_clock(),
+        )?;
+        self.command(spi, Command::MASTER_ACTIVATION)?;
+
+        self.set_sleep_mode(spi, self.sleep_mode)?;
+        Ok(())
+    }
+
+    fn update_frame(&mut self, spi: &mut SPI, buffer: &[u8]) -> Result<(), SPI::Error> {
+        assert!(buffer.len() == (WIDTH * HEIGHT / 8) as usize);
+        self.set_ram_area(spi, 0, 0, WIDTH - 1, HEIGHT - 1)?;
+        self.set_ram_address_counters(spi, 0, 0)?;
+
+        self.cmd_with_data(spi, Command::WRITE_RAM, buffer)?;
+
+        if self.refresh == RefreshLUT::FULL {
+            // Always keep the base buffer equal to current if not doing partial refresh.
+            self.set_ram_area(spi, 0, 0, WIDTH - 1, HEIGHT - 1)?;
+            self.set_ram_address_counters(spi, 0, 0)?;
+
+            self.cmd_with_data(spi, Command::WRITE_RAM_RED, buffer)?;
+        }
+        Ok(())
+    }
+
+    /// Beware that this does not work well with the screen partial refresh feature.
```

Comment:
Maybe we could also use compile_error!("Your message") to make a compile time error here if it is always better to use the different method

Reply:
Yes there is, as the partial refresh should draw more power I think (the analog and clock are left enabled) and it's not perfect. Some artifacts may be visible after several partial refreshes. In some case, it can be great, for other no so great. In my simple app, I'll use a full refresh for some events (eg. clock changing) and use partial for showing animation for alarms.